system/lib/standalone/standalone.c

```diff
@@ -124,7 +124,7 @@ size_t emscripten_get_heap_max() {
 }
 
 int emscripten_resize_heap(size_t size) {
-#if defined(EMSCRIPTEN_MEMORY_GROWTH)
+#if defined(EMSCRIPTEN_REPORT_MEMORY_GROWTH)
```

Comment:
This makes `emscripten_resize_heap()` no-op when building  with `-sSTANDALONE_WASM -sALLOW_MEMORY_GROWTH -sGROWABLE_ARRAYBUFFERS`, I think it still needs to call `__builtin_wasm_memory_grow()` in that case.

Fortunately, it looks like the `EMSCRIPTEN_PURE_WASI` macro is only used in the inner `#if` directive below, so I think you can adjust the `-pure` library variant instead (i.e. `is_pure=settings.PURE_WASI or settings.GROWABLE_ARRAYBUFFERS`).

Reply:
Hmm.. I guess we don't have enough testing for this code since I don't see failures from this obvious mistake :-/

Reply:
Indeed, there's some test coverage missing here. FWIW, this line:
https://github.com/emscripten-core/emscripten/blob/d3432e7a2a6585331c94b041973fb16c9f332ce4/src/lib/libcore.js#L2418

Should probably also be adjusted to `#if PURE_WASI && !GROWABLE_ARRAYBUFFERS`.

Reply:
But then we need a new variant of the library, which i would prefer not to add if possible.